apis/cloudflare/src/attachment-wrapper.ts

```diff
@@ -0,0 +1,165 @@
+import { isArray, isObject } from "@braintrust/core";
+import {
+  AttachmentReference,
+  braintrustAttachmentReferenceSchema,
+} from "@braintrust/core/typespecs";
+import { getBase64Parts, isBase64File } from "@braintrust/local/functions";
+import {
+  Attachment,
+  BraintrustState,
+  deserializePlainStringAsJSON,
+  ReadonlyAttachment,
+} from "braintrust";
+
+// This is a hack that makes it easy on the typesystem to deal with the payload which
+// doesn't change types.
+export function replacePayloadWithAttachments<T>(
+  data: T,
+  state: BraintrustState | undefined,
+  attachments: Record<string, AttachmentReference>,
+): T {
+  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+  return replacePayloadWithAttachmentsInner(data, state, attachments) as T;
+}
+
+function replacePayloadWithAttachmentsInner(
+  data: unknown,
+  state: BraintrustState | undefined,
+  attachments: Record<string, AttachmentReference>,
+): unknown {
+  if (isArray(data)) {
+    return data.map((item) =>
```

Comment:
#nit if you're doing an if () return, you can drop the other else if statements and avoid the tabs